packages/suite/src/views/wallet/staking/components/EthStakingDashboard/components/InstantStakeBanner.tsx

```diff
@@ -0,0 +1,121 @@
+import { Button, Icon, variables, Warning } from '@trezor/components';
+import { Translation } from 'src/components/suite';
+import styled, { useTheme } from 'styled-components';
+import { spacingsPx } from '@trezor/theme';
+import { useSelector } from 'src/hooks/suite';
+import { NetworkSymbol } from 'src/types/wallet';
+import { StakeType, WalletAccountTransaction } from '@suite-common/wallet-types';
+import { InternalTransfer } from '@trezor/connect';
+import { fromWei } from 'web3-utils';
+import { getInstantStakeType } from 'src/utils/suite/stake';
+import { isPending } from '@suite-common/wallet-utils';
+import { useEffect, useRef, useState } from 'react';
+import { selectSelectedAccount } from 'src/reducers/wallet/selectedAccountReducer';
+
+const StyledWarning = styled(Warning)`
+    justify-content: space-between;
+    font-weight: ${variables.FONT_WEIGHT.BOLD};
+    font-size: ${variables.FONT_SIZE.NORMAL};
+    padding: ${spacingsPx.xs} ${spacingsPx.lg};
+    margin-bottom: ${spacingsPx.sm};
+`;
+
+const FlexRow = styled.div`
+    display: flex;
+    align-content: space-between;
+    gap: ${spacingsPx.sm};
+`;
+
+const getTranslationId = (stakeType: StakeType) => {
+    if (stakeType === 'stake') return 'TR_STAKING_INSTANTLY_STAKED';
+
+    return 'TR_STAKING_INSTANTLY_UNSTAKED';
+};
+
+const getChangedInternalTx = (
```

Comment:
move to utils and create unit test please

Reply:
done, https://github.com/trezor/trezor-suite/pull/13267/commits/956a4043aa467b08f04408fd8607a17871383d41